src/packet_distribution.cpp

```diff
@@ -676,6 +676,7 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
         w_courtroom->arup_modify(arup_type, n_element - 1,
                                  f_contents.at(n_element));
       }
+      w_courtroom->list_areas();
```

Comment:
Unfortunately this doesn't *fully* reduce the amount of optimization possible, as the stupid ARUP system doesn't let you modify multiple data elements at once like sending CM, Players, Status etc. information all at once.
If there's a way to not even have to call list_areas() from ARUP packet, that would be peak :ok_hand: but otherwise this optimization works for now without having to rewrite the whole fucking thing.

Reply:
yeah, i was just trying to avoid rewriting too much. this reduces the number of list_areas() calls significantly, though. from about 20-30 to 4.